src/main/java/org/candlepin/subscriptions/resource/InstancesResource.java

```diff
@@ -249,12 +260,32 @@ private static List<HardwareMeasurementType> getHardwareMeasurementTypesFromCate
       ReportCategory reportCategory) {
     if (Objects.isNull(reportCategory)) {
       return new ArrayList<>();
+    } else {
+      return CATEGORY_MAP.get(reportCategory);
+    }
+  }
+
+  private static ReportCategory getCategoryByMeasurementType(
+      HardwareMeasurementType measurementType) {
+    if (HardwareMeasurementType.isSupportedCloudProvider(measurementType.name())) {
+      return ReportCategory.CLOUD;
     }
-    return switch (reportCategory) {
-      case VIRTUAL -> List.of(HardwareMeasurementType.VIRTUAL);
-      case PHYSICAL -> List.of(HardwareMeasurementType.PHYSICAL);
-      case HYPERVISOR -> List.of(HardwareMeasurementType.HYPERVISOR);
-      case CLOUD -> HardwareMeasurementType.getCloudProviderTypes();
```

Comment:
`case CLOUD` getting removed looks suspicious.  intentional or result of last merge conflict?

Reply:
I moved the check out of this switch statment so that I could use this method to check instead:
[    if (HardwareMeasurementType.isSupportedCloudProvider(measurementType.name())) 
](https://github.com/RedHatInsights/rhsm-subscriptions/pull/1815/files#diff-90f2fbedf15069ad6f8fa3491fd724676933889502bd4cab8bcf99a5c4010982R270)